code/go/internal/spectypes/contenttype.go

```diff
@@ -0,0 +1,66 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package spectypes
+
+import (
+	"encoding/json"
+	"fmt"
+	"mime"
+
+	"gopkg.in/yaml.v3"
+)
+
+// ContentType contains a content media type with its parameters.
+type ContentType struct {
+	MediaType string
+	Params    map[string]string
+}
+
+// MarshalJSON implements the json.Marshaler interface for ContentType.
+func (t ContentType) MarshalJSON() ([]byte, error) {
+	return []byte(`"` + t.String() + `"`), nil
+}
+
+// MarshalYAML implements the yaml.Marshaler interface for ContentType.
+func (t ContentType) MarshalYAML() (interface{}, error) {
+	return t.String(), nil
+}
+
+// UnmarshalJSON implements the json.Marshaler interface for ContentType.
+func (t *ContentType) UnmarshalJSON(d []byte) error {
+	var raw string
+	err := json.Unmarshal(d, &raw)
+	if err != nil {
+		return err
+	}
+
+	return t.unmarshalString(raw)
+}
+
+// UnmarshalYAML implements the yaml.Marshaler interface for ContentType.
+func (t *ContentType) UnmarshalYAML(value *yaml.Node) error {
+	// For some reason go-yaml doesn't like the UnmarshalJSON function above.
+	return t.unmarshalString(value.Value)
+}
+
+func (t *ContentType) unmarshalString(text string) error {
+	mediatype, params, err := mime.ParseMediaType(text)
+	if err != nil {
+		return err
+	}
+	if mime.FormatMediaType(mediatype, params) == "" {
+		// Bug in mime library? Happens when parsing something like "0;*0=0"
```

Comment:
I don't have any deeper knowledge of this. Did you find it by accident with unit tests?

Reply:
Gave a try to 1.18's fuzzy testing in https://github.com/elastic/package-spec/pull/278/files#diff-2a7131ad04e8aa8fcd479f359de4357a6dbe6316b61c5f6966079c4c486611f2 and found this.